src/library_addfunction.js

```diff
@@ -44,59 +44,62 @@ mergeInto(LibraryManager.library, {
     }
     return type;
   },
-
-  // Wraps a JS function as a wasm function with a given signature.
-  $convertJsFunctionToWasm__deps: ['$uleb128Encode', '$sigToWasmTypes'],
-  $convertJsFunctionToWasm: function(func, sig) {
-#if WASM2JS
-    return func;
-#else // WASM2JS
-
-    // If the type reflection proposal is available, use the new
-    // "WebAssembly.Function" constructor.
-    // Otherwise, construct a minimal wasm module importing the JS function and
-    // re-exporting it.
-    if (typeof WebAssembly.Function == "function") {
-      return new WebAssembly.Function(sigToWasmTypes(sig), func);
-    }
-
-    // The module is static, with the exception of the type section, which is
-    // generated based on the signature passed in.
-    var typeSectionBody = [
-      0x01, // count: 1
-      0x60, // form: func
-    ];
+  $generateFuncType__deps: ['$uleb128Encode'],
+  $generateFuncType : function(sig, target){
     var sigRet = sig.slice(0, 1);
     var sigParam = sig.slice(1);
     var typeCodes = {
       'i': 0x7f, // i32
-#if MEMORY64
+  #if MEMORY64
       'p': 0x7e, // i64
-#else
+  #else
       'p': 0x7f, // i32
-#endif
+  #endif
       'j': 0x7e, // i64
       'f': 0x7d, // f32
       'd': 0x7c, // f64
     };
-
+  
     // Parameters, length + signatures
-    uleb128Encode(sigParam.length, typeSectionBody);
+    target.push(0x60 /* form: func */);
+    uleb128Encode(sigParam.length, target);
     for (var i = 0; i < sigParam.length; ++i) {
-#if ASSERTIONS
+  #if ASSERTIONS
       assert(sigParam[i] in typeCodes, 'invalid signature char: ' + sigParam[i]);
-#endif
-      typeSectionBody.push(typeCodes[sigParam[i]]);
+  #endif
+  target.push(typeCodes[sigParam[i]]);
     }
-
+  
     // Return values, length + signatures
     // With no multi-return in MVP, either 0 (void) or 1 (anything else)
     if (sigRet == 'v') {
-      typeSectionBody.push(0x00);
+      target.push(0x00);
     } else {
-      typeSectionBody.push(0x01, typeCodes[sigRet]);
+      target = target.concat([0x01, typeCodes[sigRet]]);
```

Comment:
Use push here?

Any re-assignment to `target` in the function won't work right?  (since its an incoming param)